test/FullRange.t.sol

```diff
@@ -0,0 +1,873 @@
+            Currency.wrap(address(token0)), Currency.wrap(address(token1)), 0, TICK_SPACING + 1, fullRange
+        );
+
+        vm.expectRevert("Tick spacing must be default");
+        manager.initialize(wrongKey, SQRT_RATIO_1_1);
+    }
+
+    function gasTestInitialize() public {
+        snapStart("initialize no fee");
+        manager.initialize(key, SQRT_RATIO_1_1);
+        snapEnd();
+    }
+
+    function gasTestInitializeFee() public {
+        snapStart("initialize with fee");
+        manager.initialize(feeKey, SQRT_RATIO_1_1);
+        snapEnd();
+    }
+
+    function testInitialAddLiquiditySucceeds() public {
+        manager.initialize(key, SQRT_RATIO_1_1);
+
+        uint256 currBalance0 = TestERC20(token0).balanceOf(address(this));
+        uint256 currBalance1 = TestERC20(token1).balanceOf(address(this));
+
+        snapStart("add liquidity");
+        fullRange.addLiquidity(address(token0), address(token1), 0, 100, 100, address(this), 12329839823);
+        snapEnd();
+
+        assertEq(TestERC20(token0).balanceOf(address(this)), currBalance0 - 100);
+        assertEq(TestERC20(token1).balanceOf(address(this)), currBalance1 - 100);
+
+        (,,,,,, address liquidityToken) = fullRange.poolToInfo(id);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 100);
+    }
+
+    function testInitialAddLiquidityWithFeeSucceeds() public {
+        manager.initialize(feeKey, SQRT_RATIO_1_1);
+
+        uint256 currBalance0 = TestERC20(token0).balanceOf(address(this));
+        uint256 currBalance1 = TestERC20(token1).balanceOf(address(this));
+
+        snapStart("add liquidity with fee");
+        fullRange.addLiquidity(address(token0), address(token1), 3000, 100, 100, address(this), 12329839823);
+        snapEnd();
+
+        assertEq(TestERC20(token0).balanceOf(address(this)), currBalance0 - 100);
+        assertEq(TestERC20(token1).balanceOf(address(this)), currBalance1 - 100);
+
+        (,,,,,, address liquidityToken) = fullRange.poolToInfo(feeId);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 100);
+
+        // check pool position state
+        (
+            uint128 liquidity,
+            uint256 feeGrowthInside0LastX128,
+            uint256 feeGrowthInside1LastX128,
+            uint128 tokensOwed0,
+            uint128 tokensOwed1,
+            ,
+        ) = fullRange.poolToInfo(feeId);
+
+        assertEq(liquidity, 100);
+        // TODO: make sure 0 is correct
+        assertEq(feeGrowthInside0LastX128, 0);
+        assertEq(feeGrowthInside1LastX128, 0);
+        assertEq(tokensOwed0, 0);
+        assertEq(tokensOwed1, 0);
+    }
+
+    function testAddLiquidityFailsIfNoPool() public {
+        // PoolNotInitialized()
+        vm.expectRevert(0x486aa307);
+        fullRange.addLiquidity(address(token0), address(token1), 0, 100, 100, address(this), 12329839823);
+    }
+
+    function testAddLiquiditySucceedsWithNoFee() public {
```

Comment:
as mentioned above, thinking these `WithNoFee` and `WithFee` tests are redundant. Just test WithFee (not needed in test case) and only test noFee for when it's necessary to make sure this edgecase works in certain scenarios. Unless you have a reason for testing it so much already??